pkg/metrics/timer.go

```diff
@@ -23,6 +23,9 @@ func (t *Timer) Update(d time.Duration) { atomic.StoreInt64(&t.val, int64(d)) }
 // UpdateSince sets the timer value to the difference between since and the current time.
 func (t *Timer) UpdateSince(since time.Time) { t.Update(time.Since(since)) }
 
+// UpdateSinceAccumulate adds to the timer's current value the difference between since and the current time.
+func (t *Timer) UpdateSinceAccumulate(since time.Time) { t.Update(time.Since(since) + t.Value()) }
```

Comment:
This should use the `atomic` functions to avoid a potential race condition.

```go
func (t *Timer) UpdateSinceAccumulate(since time.Time) { atomic.AddIn64(&t.val, time.Since(since).Nanoseconds()) }
```

I'd also be tempted to change `Timer.val` to an `atomic.Int64` while we're touching this file.

Reply:
done. thanks for the good catch